python/pyarrow/pandas_compat.py

```diff
@@ -449,6 +455,19 @@ def _get_columns_to_convert_given_schema(df, schema, preserve_index):
             index_levels, columns_to_convert, convert_fields)
 
 
+def _get_index_level(df, name):
+    """
+    Get the index level of a DataFrame given 'name' (column name in an arrow
+    Schema).
+    """
+    key = name
+    if name not in df.index.names and _is_generated_index_name(name):
+        # we know we have an autogenerated name => extract number and get
+        # the index level positionally
+        key = int(name.lstrip("__index_level_").rstrip("__"))
```

Comment:
This isn't terrific:
```python
>>> "xxx".lstrip("__index_level_")
''
```

Reply:
The check in the if statement (`_is_generated_index_name(name)`) ensures it has the `__index_level_i__` format. I could use a `re.match` to get out the `i`, but because I know the exact format, that felt like unnecessary overhead.

Reply:
Or simply `name[len("__index_level_"):-2]`?

Reply:
Thanks, that's indeed a bit simpler. Updated with this.